pages/_app.tsx

```diff
@@ -4,22 +4,48 @@ import { theme } from "../theme/theme";
 import WalletWrapper from "../components/Wallet/WalletWrapper";
 import Layout from "../components/Layout";
 
+import {
+  ConnectionProvider,
+  WalletProvider,
+} from "@solana/wallet-adapter-react";
+import { useMemo } from "react";
+import { clusterApiUrl } from "@solana/web3.js";
+import dynamic from "next/dynamic";
+import {
+  GlowWalletAdapter,
+  PhantomWalletAdapter,
+  SolflareWalletAdapter,
+  TorusWalletAdapter,
+} from "@solana/wallet-adapter-wallets";
+
 export default function App({ Component, pageProps }: AppProps) {
-  const endpoint = "https://api.devnet.solana.com";
+  const wallets = useMemo(
+    () => [
+      new PhantomWalletAdapter(),
+      new SolflareWalletAdapter(),
+      new GlowWalletAdapter(),
+      new TorusWalletAdapter(),
+    ],
+    []
+  );
+  // const endpoint = "https://api.devnet.solana.com";
+  const rpc = useMemo(() => clusterApiUrl("mainnet-beta"), []);
 
   return (
     <div>
-      <ChakraProvider theme={theme}>
-        <WalletWrapper endpoint={endpoint}>
-          {typeof window !== undefined ? (
-            <Layout>
-              <Component {...pageProps} />
-            </Layout>
-          ) : (
-            <Text>Undefined</Text>
-          )}
-        </WalletWrapper>
-      </ChakraProvider>
+      <ConnectionProvider endpoint={rpc}>
+        <ChakraProvider theme={theme}>
+          <WalletProvider wallets={wallets} autoConnect>
+            {typeof window !== undefined ? (
+              <Layout>
+                <Component {...pageProps} />
+              </Layout>
+            ) : (
+              <Text>Undefined</Text>
+            )}
+          </WalletProvider>
+        </ChakraProvider>
+      </ConnectionProvider>
     </div>
```

Comment:
The logic is being repeated and i dont know why you have removed walletWrapper all this logic could be done there

Reply:
I was getting some trouble doing it with Walletwrapper but I guess I have figured that out. I will try to use WalletWrapper rather than writing a whole new logic.

Reply:
> The logic is being repeated and i dont know why you have removed walletWrapper all this logic could be done there